From: Thomas Moulard <thomas.moulard@gmail.com>
Date: Wed, 5 Dec 2012 11:47:36 +0900
Subject: Fix compilation failure with recent Boost version.

intrusive_ptr_add_ref and intrusive_ptr_release should be in the
class namespace and not boost namespace.

See https://svn.boost.org/trac/boost/ticket/6943

Forwarded: yes
Author: Thomas Moulard <thomas.moulard@gmail.com>
---
 src/Util/Referenced.h | 8 ++++----
 src/Util/YamlNodes.h  | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/Util/Referenced.h b/src/Util/Referenced.h
index 7e800e4..4b073b3 100644
--- a/src/Util/Referenced.h
+++ b/src/Util/Referenced.h
@@ -11,7 +11,7 @@ namespace cnoid {
     class Referenced;
 }
 
-namespace boost{
+namespace cnoid {
     void intrusive_ptr_add_ref(cnoid::Referenced* obj);
     void intrusive_ptr_release(cnoid::Referenced* obj);
 }
@@ -28,8 +28,8 @@ namespace cnoid {
         inline int refCounter() { return refCounter_; }
 
       private:
-        friend void boost::intrusive_ptr_add_ref(Referenced* obj);
-        friend void boost::intrusive_ptr_release(Referenced* obj);
+        friend void intrusive_ptr_add_ref(Referenced* obj);
+        friend void intrusive_ptr_release(Referenced* obj);
 
         int refCounter_;
     };
@@ -37,7 +37,7 @@ namespace cnoid {
     typedef boost::intrusive_ptr<Referenced> ReferencedPtr;
 }
 
-namespace boost
+namespace cnoid
 {
     inline void intrusive_ptr_add_ref(cnoid::Referenced* obj){
         obj->refCounter_++;
diff --git a/src/Util/YamlNodes.h b/src/Util/YamlNodes.h
index 74e99f8..3f95cc4 100644
--- a/src/Util/YamlNodes.h
+++ b/src/Util/YamlNodes.h
@@ -17,7 +17,7 @@ namespace cnoid {
     class YamlNode;
 }
 
-namespace boost{
+namespace cnoid {
     void intrusive_ptr_add_ref(cnoid::YamlNode* obj);
     void intrusive_ptr_release(cnoid::YamlNode* obj);
 }
@@ -161,8 +161,8 @@ namespace cnoid {
         friend class YamlMapping;
         friend class YamlSequence;
 
-        friend void boost::intrusive_ptr_add_ref(YamlNode* obj);
-        friend void boost::intrusive_ptr_release(YamlNode* obj);
+        friend void intrusive_ptr_add_ref(YamlNode* obj);
+        friend void intrusive_ptr_release(YamlNode* obj);
     };
 
     typedef boost::intrusive_ptr<YamlNode> YamlNodePtr;
@@ -508,7 +508,7 @@ namespace cnoid {
 }
 
 
-namespace boost
+namespace cnoid
 {
     inline void intrusive_ptr_add_ref(cnoid::YamlNode* obj){
         obj->refCounter++;
